src/dmd/dcast.d

```diff
@@ -3453,6 +3453,40 @@ Expression typeCombine(BinExp be, Scope* sc)
             return errorReturn();
     }
 
+    if (sc && sc.flags & SCOPE.Cfile)
+    {
+        // https://issues.dlang.org/show_bug.cgi?id=22275
+        // Static arrays saturate to pointers 
+        if (t1.ty == Tsarray || t1.ty == Tarray)
+        {
+            be.e1 = new DotIdExp(be.loc, be.e1, Id.ptr).expressionSemantic(sc);
+            t1 = be.e1.type.toBasetype();
+        }
+        if (t2.ty == Tsarray || t2.ty == Tarray)
+        {
+            be.e2 = new DotIdExp(be.loc, be.e2, Id.ptr).expressionSemantic(sc);
+            t2 = be.e2.type.toBasetype();
+        }
```

Comment:
Use `arrayFuncConv()` instead, which already does this (and also handles function conversion to pointer).